### PR TITLE
fix: use regional terrain for the car showroom

### PR DESCRIPTION
**Pull request by 0Monet**
### Required

- [X] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [X] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [X] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [X] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change
Use regional terrain for the car showroom.
## Describe the solution
Replace non-regional terrain with regional terrain.
Use roof_palette for the mapgen `cs_car_showroom_roof`.
## Describe alternatives you've considered
none
## Additional context
Before:
image1
image2
After:
image3
image4

### CODE REVIEW

File: data/json/mapgen/cs_car_showroom.json [hunk 1]
```diff
@@ -3,51 +3,49 @@
     "type": "mapgen",
     "om_terrain": [ "cs_car_showroom" ],
     "method": "json",
-    "weight": 1000,
```
Comment (scarf005) [inline data/json/mapgen/cs_car_showroom.json:6]: is this intended? if so, could you explain what it does? (i'm not familiar with mapgen)

**Review by scarf005 — APPROVED** [review]

File: data/json/mapgen/cs_car_showroom.json (hunk 1 above)
Comment (0Monet) [reply data/json/mapgen/cs_car_showroom.json:6]: "weight" is useful if, for example, there are two or more mapgen with an identical id, the one with the highest "weight" will be chosen more often than the one with the lowest "weight" on placement.

Comment (scarf005) [reply data/json/mapgen/cs_car_showroom.json:6]: ...oh, so it got removed here because there's only one car showroom.

Comment (0Monet) [reply data/json/mapgen/cs_car_showroom.json:6]: Yes.

**Pull request merged**